tests/Doctrine/Tests/ORM/Functional/Ticket/GH7684Test.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\ORM\Mapping\MappingException;
+use Doctrine\Tests\ORM\Functional\DatabaseDriverTestCase;
+
+/**
+ * Verifies that associations/columns with an inline '_id' get named properly
+ * 
+ * Github issue: 7684
+ */
+class GH7684 extends DatabaseDriverTestCase
+{
+    public function testIssue() : void
+    {
+        if (!$this->_em->getConnection()->getDatabasePlatform()->supportsForeignKeyConstraints()) {
+            $this->markTestSkipped('Platform does not support foreign keys.');
+        }
+
+        $card = new Table('partner_identity_card');
```

Comment:
Could we drop the domain naming around this, and use test-specific naming for columns/tables? This would remove focus from your specific domain, and onto the issue details

Reply:
Yeah, removed the domain specific stuff